src/services/RegisterService.js

```diff
@@ -1,17 +1,28 @@
 import http from './HttpService';
+import profile from './CreateProfile';
 
 const registration = {
-    register: function (name, email, password, chooseFile) {
-        http.post('/auth/local/register', {
-            name: name,
+    register: async function (name, email, password, file) {
+        const response = await http.post('/auth/local/register', {
+            username: name,
             email: email,
             password: password,
-            chooseFile: chooseFile.current.value,
-        })
-            .then((res) => {
-                console.log(res);
-            })
-            .catch((error) => console.log(error.message));
+        });
+
+        const token = response.data.jwt;
+        localStorage.setItem('token', token);
+
+        const responseUpload = await http.post('/upload', file, {
+            headers: {
+                Authorization: `Bearer ${token}`,
+            },
+        });
+
+        const username = response.data.user.username;
+        const userId = response.data.user.id;
+        const profilePhotoId = responseUpload.data[0].id;
+
+        profile.createProfile(username, userId, profilePhotoId, '4');
```

Comment:
We should add `await` here too, so we know when all ajax calls in this function are done